### PR TITLE
Added visibility flag to Organization create_repo

### PR DESCRIPTION
**Pull request by ruarfff**
Update for issue #1446

### CODE REVIEW

File: github/Organization.py [hunk 1]
```diff
@@ -577,9 +582,18 @@ def create_repo(
...
             post_parameters["delete_branch_on_merge"] = delete_branch_on_merge
+        if visibility is not github.GithubObject.NotSet:
+            post_parameters["visibility"] = visibility
+
         headers, data = self._requester.requestJsonAndCheck(
-            "POST", self.url + "/repos", input=post_parameters
+            "POST",
+            self.url + "/repos",
+            input=post_parameters,
+            headers={"Accept": Consts.repositoryVisibilityPreview}
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:592]: Or set headers when you check is visibility isn't NotSet?

**Comment by ruairitobrien** [comment]
> I like the direction, but I also don't think it goes far enough. We should also change get_repos/get_repo calls to send the header so we can populate visibility on the Repository object, which we should also support.

Makes sense. Will do that.

File: github/Organization.py (hunk 1 above)
Comment (ruairitobrien) [reply github/Organization.py:592]: Do you mean to invert the condition? Like

```
headers=None if visibility is github.GithubObject.NotSet else  {"Accept": Consts.repositoryVisibilityPreview}
```

Of do you mean to initialise `headers, data = self._requester.requestJsonAndCheck` conditionally?

My apologies, I am fairly new to python so might be missing something obvious here  :)

File: github/Organization.py [hunk 2]
```diff
@@ -544,6 +546,9 @@ def create_repo(
         assert delete_branch_on_merge is github.GithubObject.NotSet or isinstance(
             delete_branch_on_merge, bool
         ), delete_branch_on_merge
+        assert visibility is github.GithubObject.NotSet or isinstance(
+            visibility, str
+        ), visibility
         post_parameters = {
             "name": name,
         }
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:554]: So up here, I'd say something like: ``post_headers = dict()``

File: github/Organization.py [hunk 3]
```diff
@@ -577,9 +582,18 @@ def create_repo(
             post_parameters["allow_rebase_merge"] = allow_rebase_merge
         if delete_branch_on_merge is not github.GithubObject.NotSet:
             post_parameters["delete_branch_on_merge"] = delete_branch_on_merge
+        if visibility is not github.GithubObject.NotSet:
+            post_parameters["visibility"] = visibility
+
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:587]: Then here, you can add the const to ``post_headers``.

File: github/Organization.py (hunk 1 above)
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:592]: And then down here, you just have ``headers=post_headers,``

And also, no fair apologising for learning! Python is fun, and I'm always delighted to help people learn one of my favourite languages.

File: github/Organization.py [hunk 4]
```diff
@@ -577,9 +582,18 @@ def create_repo(
             post_parameters["allow_rebase_merge"] = allow_rebase_merge
         if delete_branch_on_merge is not github.GithubObject.NotSet:
             post_parameters["delete_branch_on_merge"] = delete_branch_on_merge
+        if visibility is not github.GithubObject.NotSet:
+            post_parameters["visibility"] = visibility
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:586]: Thinking about it, we may also want assert on the content of visibility, such as this example from git grep:

```
github/Repository.py:        assert state in ["open", "closed"], state
```

File: github/Organization.py [hunk 5]
```diff
@@ -477,6 +477,7 @@ def create_repo(
         allow_merge_commit=github.GithubObject.NotSet,
         allow_rebase_merge=github.GithubObject.NotSet,
         delete_branch_on_merge=github.GithubObject.NotSet,
+        visibility=github.GithubObject.NotSet,
```
Comment (s-t-e-v-e-n-k) [inline github/Organization.py:480]: You should also edit the github/Organization.pyi file to add the new parameter.

Comment (ruairitobrien) [reply github/Organization.py:480]: Ah yes. Thank you!

File: github/Organization.py (hunk 1 above)
Comment (ruairitobrien) [reply github/Organization.py:592]: Makes sense. Thanks

File: tests/Organization.py [hunk 6]
```diff
@@ -324,13 +326,18 @@ def testCreateRepoWithAllArguments(self):
...
+            visibility="internal",
         )
         self.assertEqual(
             repo.url, "https://api.github.com/repos/BeaverSoftware/TestPyGithub2"
         )
         self.assertFalse(repo.has_wiki)
         self.assertFalse(repo.has_pages)
 
+    def testCreateRepoWithInvalidVisibility(self):
+        with pytest.raises(AssertionError):
```
Comment (s-t-e-v-e-n-k) [inline tests/Organization.py:338]: ``with self.assertRaises()`` instead of importing pytest.

Comment (ruairitobrien) [reply tests/Organization.py:338]: Perfect. Thanks!

**Comment by ruairitobrien** [comment]
> Just one niggle left, but this is looking good. I'm tossing up whether we add visibility to get_repos() in this PR or a different one.

@s-t-e-v-e-n-k 

Sorry it took me a while to get back to this. 

I am looking at adding the header to get_repos in various places. One issue I am hitting again is getting the recorded tests to pass. 

For example, I went into Organization.get_repos and added this:

```python
# Top of class
VISIBILITY_PREVIEW_HEADERS = {"Accept": Consts.repositoryVisibilityPreview}

# In the get_repos method
return github.PaginatedList.PaginatedList(
            github.Repository.Repository,
            self._requester,
            self.url + "/repos",
            url_parameters,
            headers=VISIBILITY_PREVIEW_HEADERS,
        )
```

When I run `pytest` I see 7 failures in various test cases. 

```
============================================================================================= short test summary info ==============================================================================================
FAILED tests/Authentication.py::Authentication::testSecretKeyAuthentication - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Issue158.py::Issue158::testPaginationWithSecretKeyAuthentication - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Issue80.py::Issue80::testIgnoreHttpsFromGithubEnterprise - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Issue80.py::Issue80::testIgnoreHttpsFromGithubEnterpriseWithPort - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Organization.py::Organization::testGetRepos - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Organization.py::Organization::testGetReposSorted - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
FAILED tests/Organization.py::Organization::testGetReposWithType - requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))
ERROR tests/Authentication.py::Authentication::testSecretKeyAuthentication - AssertionError: 'None' != ''
ERROR tests/Issue158.py::Issue158::testPaginationWithSecretKeyAuthentication - AssertionError: 'None' != ''
ERROR tests/Issue80.py::Issue80::testIgnoreHttpsFromGithubEnterprise - AssertionError: 'None' != ''
ERROR tests/Issue80.py::Issue80::testIgnoreHttpsFromGithubEnterpriseWithPort - AssertionError: 'None' != ''
ERROR tests/Organization.py::Organization::testGetRepos - AssertionError: 'None' != ''
ERROR tests/Organization.py::Organization::testGetReposSorted - AssertionError: 'None' != ''
ERROR tests/Organization.py::Organization::testGetReposWithType - AssertionError: 'None' != ''
===================================================================================== 7 failed, 634 passed, 7 errors in 37.49s =====================================================================================
```

I've been trying to follow the instructions in CONTRiBUTION.md to see if I can update the recording to include the new header. I created a personal access token with most permissions and tried:

```
pytest tests/Organization.py -k testGetRepos --record --auth_with_token
```

Failure output:

```
FAILED tests/Organization.py::Organization::testGetRepos - github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://developer.github.com/v3/orgs/#get-an-organiz...
FAILED tests/Organization.py::Organization::testGetReposSorted - github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://developer.github.com/v3/orgs/#get-an-o...
FAILED tests/Organization.py::Organization::testGetReposWithType - github.GithubException.UnknownObjectException: 404 {"message": "Not Found", "documentation_url": "https://developer.github.com/v3/orgs/#get-an...
```

I know if I go here I get a 404 https://github.com/BeaverSoftware but I suspect the issue is I am completely misunderstanding how the tests work. 

I'll keep at it trying to figure out how it works but if you have any tips please let me know :)

**Pull request closed without merging**